### PR TITLE
Gson Serializer/Deserializer are now correctly registered service when found on the classpath

### PR DESCRIPTION
**Pull request by bdemers**
Added test to other serializer implementations as well

Fixes: #563

### CODE REVIEW

**Comment by lhazlewood** [comment]
I think we need to target this to a new 0.11.x branch (not master) since this will go into the 0.11.1 release, no?

File: extensions/gson/src/test/groovy/io/jsonwebtoken/gson/io/GsonDeserializerTest.groovy [hunk 1]
```diff
@@ -17,14 +17,22 @@ package io.jsonwebtoken.gson.io
 
 import com.google.gson.Gson
 import io.jsonwebtoken.io.DeserializationException
+import io.jsonwebtoken.io.Deserializer
 import io.jsonwebtoken.lang.Strings
+import org.hamcrest.CoreMatchers
```
Comment (lhazlewood) [inline extensions/gson/src/test/groovy/io/jsonwebtoken/gson/io/GsonDeserializerTest.groovy]: we use EasyMock's API everywhere else in the project - can we please stick with those?

File: extensions/gson/src/test/groovy/io/jsonwebtoken/gson/io/GsonSerializerTest.groovy [hunk 2]
```diff
@@ -15,16 +15,27 @@
  */
 package io.jsonwebtoken.gson.io
 
+import io.jsonwebtoken.io.Deserializer
+import io.jsonwebtoken.io.Serializer
 import io.jsonwebtoken.lang.Strings
+import org.hamcrest.CoreMatchers
```
Comment (lhazlewood) [inline extensions/gson/src/test/groovy/io/jsonwebtoken/gson/io/GsonSerializerTest.groovy]: Same (EasyMock)

File: extensions/jackson/src/test/groovy/io/jsonwebtoken/jackson/io/JacksonDeserializerTest.groovy [hunk 3]
```diff
@@ -17,16 +17,23 @@ package io.jsonwebtoken.jackson.io
 
 import com.fasterxml.jackson.databind.ObjectMapper
 import io.jsonwebtoken.io.DeserializationException
+import io.jsonwebtoken.io.Deserializer
 import io.jsonwebtoken.io.Encoders
 import io.jsonwebtoken.jackson.io.stubs.CustomBean
 import io.jsonwebtoken.lang.Maps
 import io.jsonwebtoken.lang.Strings
+import org.hamcrest.CoreMatchers
```
Comment (lhazlewood) [inline extensions/jackson/src/test/groovy/io/jsonwebtoken/jackson/io/JacksonDeserializerTest.groovy]: EasyMock

File: extensions/jackson/src/test/groovy/io/jsonwebtoken/jackson/io/JacksonSerializerTest.groovy [hunk 4]
```diff
@@ -18,14 +18,22 @@ package io.jsonwebtoken.jackson.io
 import com.fasterxml.jackson.core.JsonProcessingException
 import com.fasterxml.jackson.databind.ObjectMapper
 import io.jsonwebtoken.io.SerializationException
+import io.jsonwebtoken.io.Serializer
 import io.jsonwebtoken.lang.Strings
+import org.hamcrest.CoreMatchers
```
Comment (lhazlewood) [inline extensions/jackson/src/test/groovy/io/jsonwebtoken/jackson/io/JacksonSerializerTest.groovy]: EasyMock :)

File: extensions/orgjson/src/test/groovy/io/jsonwebtoken/orgjson/io/OrgJsonDeserializerTest.groovy [hunk 5]
```diff
@@ -16,13 +16,20 @@
 package io.jsonwebtoken.orgjson.io
 
 import io.jsonwebtoken.io.DeserializationException
+import io.jsonwebtoken.io.Deserializer
 import io.jsonwebtoken.lang.Strings
-import io.jsonwebtoken.orgjson.io.OrgJsonDeserializer
+import org.hamcrest.CoreMatchers
```
Comment (lhazlewood) [inline extensions/orgjson/src/test/groovy/io/jsonwebtoken/orgjson/io/OrgJsonDeserializerTest.groovy]: idem

File: extensions/orgjson/src/test/groovy/io/jsonwebtoken/orgjson/io/OrgJsonSerializerTest.groovy [hunk 6]
```diff
@@ -17,9 +17,10 @@ package io.jsonwebtoken.orgjson.io
 
 import io.jsonwebtoken.SignatureAlgorithm
 import io.jsonwebtoken.io.SerializationException
+import io.jsonwebtoken.io.Serializer
 import io.jsonwebtoken.lang.DateFormats
 import io.jsonwebtoken.lang.Strings
-import io.jsonwebtoken.orgjson.io.OrgJsonSerializer
+import org.hamcrest.CoreMatchers
```
Comment (lhazlewood) [inline extensions/orgjson/src/test/groovy/io/jsonwebtoken/orgjson/io/OrgJsonSerializerTest.groovy]: idem

**Comment by lhazlewood** [comment]
Sorry, in the review above, where I said `EasyMock`, I meant `org.junit.Assert.*`

**Comment by bdemers** [comment]
@lhazlewood back to you, I just squashed & rebased

**Pull request merged**